libraries/engage/product/helpers/index.js

```diff
@@ -44,3 +43,41 @@ export const setDefaultProductFetchParams = () => {
   });
 };
 
+/**
+ * Provides the settings for ProductImages
+ * @return {Object}
+ */
+export const getProductImageSettings = () => {
```

Comment:
This can become a react hook, with `useMemo` inside to memoize calculation of settings